Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -113,47 +138,60 @@ - (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
         // We only create a span if there is a transaction in the scope,
         // otherwise we have nothing else to do here.
         if (netSpan == nil) {
-            [SentryLog
-                logWithMessage:@"No transaction bound to scope. Won't track network operation."
-                      andLevel:kSentryLevelDebug];
-
+            SENTRY_LOG_DEBUG(@"No transaction bound to scope. Won't track network operation.");
             return;
         }
 
-        if ([sessionTask currentRequest]) {
-            NSMutableURLRequest *newRequest = nil;
+        if ([sessionTask currentRequest] &&
+            [self addHeadersForRequestWithURL:sessionTask.currentRequest.URL]) {
+            NSString *baggageHeader = @"";
+
+            SentryTracer *tracer = [SentryTracer getTracer:span];
+            if (tracer != nil) {
+                baggageHeader = [[tracer.traceContext toBaggage]
+                    toHTTPHeaderWithOriginalBaggage:
+                        [SentrySerialization
+                            decodeBaggage:sessionTask.currentRequest
+                                              .allHTTPHeaderFields[SENTRY_BAGGAGE_HEADER]]];
+            }
 
             // First we check if the current request is mutable, so we could easily add a new
             // header. Otherwise we try to change the current request for a new one with the extra
             // header.
             if ([sessionTask.currentRequest isKindOfClass:[NSMutableURLRequest class]]) {
-                newRequest = (NSMutableURLRequest *)sessionTask.currentRequest;
-                [newRequest setValue:[netSpan toTraceHeader].value
-                    forHTTPHeaderField:SENTRY_TRACE_HEADER];
+                NSMutableURLRequest *currentRequest
+                    = (NSMutableURLRequest *)sessionTask.currentRequest;
+                [currentRequest setValue:[netSpan toTraceHeader].value
+                      forHTTPHeaderField:SENTRY_TRACE_HEADER];
+
+                [currentRequest setValue:baggageHeader forHTTPHeaderField:SENTRY_BAGGAGE_HEADER];
```

Comment:
`m`: I think we don't need to add the header if it's empty, which could happen if the `tracer == nil` above.